api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -241,6 +241,20 @@ public void setFirstName(final String firstName) {
         this.firstName = firstName;
     }
 
+    @Nullable
+    public String getRawPreferredName() {
+        return preferredName;
+    }
+
+    @Nullable
+    public String getPreferredName() {
+        return !Strings.isNullOrEmpty(preferredName) ? preferredName : firstName;
+    }
+
+    public void setPreferredName(@Nullable final String name) {
+        preferredName = Objects.equal(firstName, name) ? null : name;
```

Comment:
Why `"internally only track preferredNames that are different from firstName"` and not simply:
```
public void setPreferredName(@Nullable final String preferredName) {
   this.preferredName = preferredName;
}
```